nntrainer/include/databuffer_v2.h

```diff
@@ -0,0 +1,325 @@
+// SPDX-License-Identifier: Apache-2.0-only
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	databuffer_v2.h
+ * @date	3 September 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is databuffer class for Neural Network
+ *
+ * @todo TODO: Support multi files for dataset with files
+ * @todo TODO: Support multi threads with more than 1 thread and use thread
+ * pooling
+ * @todo TODO: Support label size to be 0 for inference based scenarios
+ * @todo TODO: rename data buffer to dataset
+ * @todo TODO: manage with just 1 buffer
+ * @todo TODO: consider different data structure for buffer when using shuffle
+ */
+
+#ifndef __DATABUFFER_V2_H__
+#define __DATABUFFER_V2_H__
+#ifdef __cplusplus
+
+#include <condition_variable>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <tuple>
+#include <vector>
+
+#include <nntrainer-api-common.h>
+#include <tensor.h>
+
+namespace nntrainer {
+
+/**
+ * @brief   States the collect thread running in the background
+ */
+enum class ThreadStates {
+  THREAD_NULL,            /**< not yet initialized */
+  THREAD_READY,           /**< initialized but not yet started */
+  THREAD_RUNNING,         /**< started and running */
+  THREAD_REQUEST_TO_STOP, /**< main thread has request to stop, background
```

Comment:
Just a suggestion. `THREAD_DONE` ? `THREAD_JOINABLE` ?

Reply:
how aobut `THREAD_REQUESTED_TO_STOP`. IMO these are more verbose and reader understands what it means.
THREAD_DONE/THREAD_JOINABLE means thread is finished. THREAD_WAITING is too generic.

Reply:
Got the meaning. let's not change.